src/app/qgisapp.cpp

```diff
@@ -7030,6 +7030,12 @@ void QgisApp::changeDataSource( QgsMapLayer *layer )
           tl->setItemVisibilityChecked( true );
         }
       }
+
+      // Tell the bridge that we have fixed a layer
+      if ( ! layerWasValid && layer->isValid() )
+      {
+        QgsProject::instance()->layerTreeRoot()->customLayerOrderChanged( );
```

Comment:
my ignorance, why customLayerOrderChanged and not a more generic layerOrderChanged?